### PR TITLE
Add more SGX tasks

### PR DESCRIPTION
**Pull request by csegarragonz**
In this PR I add a couple of tasks to interact with the attestation provider and get/set attestation policies.

I also move the tasks to provision and delete an SGX-enabled VM, which previously  lived in a branch of `experiment-base`. After thinking about it I feel it fits here better.

### CODE REVIEW

File: tasks/sgx.py [hunk 1]
```diff
@@ -15,7 +15,10 @@ def do_configure(repo_url, write_file, key_url):
 
 
 @task
-def configure():
+def configure(ctx):
```
Comment (csegarragonz) [inline tasks/sgx.py:22]: This file was comitted before, feel free to leave comments on any line, even if it is not part of this `diff`.

File: tasks/sgx.py [hunk 2]
```diff
@@ -15,7 +15,10 @@ def do_configure(repo_url, write_file, key_url):
...
+    """
+    Configure Intel and Microsoft APT repos
+    """
```
Comment (Shillaker) [inline tasks/sgx.py:25]: This is convenient but looks like it's ubuntu 18.04-specific, might be worth putting a warning in the `README` about OS and version dependencies, then including a link to the official instructions for anyone who wants to diverge from that.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # SGX Experiments in Faasm
 
+This repository contains instructions to run SGX experiments on Faasm.
+For the moment, though, it contains instructions to provision an SGX-enabled
+VM and test that it can interact with the attestation service in Azure.
+
```
Comment (Shillaker) [inline README.md:6]: I know we don't really enforce style on `.md` files but in the rest of the repos we put a linebreak before and after code snippets. Doesn't change the presentation of the rendered markdown but it makes the source easier to read.

File: tasks/sgx.py [hunk 4]
```diff
@@ -81,45 +94,50 @@ def install_sgxsdk():
...
         install_sgxsdk()
     install_net_core_sdk()
     clone_azure_attestation_repo()
 
 
-@task
-def generate_quotes():
+def demo_generate_quotes():
     run(
         "cd /opt/maa/intel.sdk.attest.sample/genquotes && AZDCAP_DEBUG_LOG_LEVEL=INFO bash ./runall.sh",
```
Comment (Shillaker) [inline tasks/sgx.py]: Rather than doing `cd &&` in these commands you can specify a working directory with the `cwd` kwarg.

**Review by Shillaker — CHANGES_REQUESTED** [review]
Just a few smaller things. Looks pretty gnarly 😬

**Review by Shillaker — APPROVED** [review]

**Pull request merged**